analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsPattern.cs

```diff
@@ -70,9 +70,9 @@ private static SymbolicConstraint ConstraintFromConstantPattern(ProgramState sta
 
     private static ObjectConstraint ConstraintFromRecursivePattern(IRecursivePatternOperationWrapper recursive, bool falseBranch, bool hasObjectConstraint)
     {
-        if (hasObjectConstraint || !recursive.InputType.IsReferenceType)
+        if (hasObjectConstraint)
```

Comment:
This removal may set the NullConstraint for a non-nullable value type symbol in the following elseif branch, but that gets reverted by the post check "NonNullableValueTypeCheck". It might be helpful to add a comment on line 49 about this behavior.

Reply:
"reverted"? I'd say it can get re-applied. And that will result in no operation in `NonNullableValueTypeCheck`.

No harm should be done.

Reply:
I'm talking about line79: `return RecursivePatternAlwaysMatchesAnyNotNull(recursive) ? ObjectConstraint.Null : null;` Here the NullConstraint might be set for a non-nullable value type now.